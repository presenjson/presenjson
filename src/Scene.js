import React from 'react';
import PropTypes from 'prop-types';
import Style from './Style';

const Scene = ({ children, length, ...rest }) => (
    <Style {...rest} scene>
        {children}
    </Style>
);

Scene.propTypes = {
    length: PropTypes.number,
    children: PropTypes.node
};

Scene.defaultProps = {
    length: 0
};

export default Scene;
