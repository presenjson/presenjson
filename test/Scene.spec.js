import React from 'react';
import { shallow } from 'enzyme';
import Scene from '../src/Scene';

describe('Scene', () => {
    it('renders correctly', () => {
        expect(
            shallow(
                <Scene light>
                    <div className="test" />
                </Scene>
            )
        ).toMatchSnapshot();
    });
});
