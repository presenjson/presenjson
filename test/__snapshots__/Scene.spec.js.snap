// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scene renders correctly 1`] = `
<Style
  light={true}
  scene={true}
>
  <div
    className="test"
  />
</Style>
`;
